app/views/courses/HeroSelectModal.coffee

```diff
@@ -0,0 +1,42 @@
+ModalView = require 'views/core/ModalView'
+template = require 'templates/courses/hero-select-modal'
+Classroom = require 'models/Classroom'
+CocoCollection = require 'collections/CocoCollection'
+State = require 'models/State'
+ThangType = require 'models/ThangType'
+User = require 'models/User'
+
+module.exports = class HeroSelectModal extends ModalView
+  id: 'hero-select-modal'
+  template: template
+
+  events:
+    'click .select-hero-btn': 'onClickSelectHeroButton'
+    'click .hero-option': 'onClickHeroOption'
+
+  initialize: ({ currentHeroID }) ->
+    @state = new State({
+      currentHeroID
+      selectedHeroID: currentHeroID
+    })
+
+    @heroes = new CocoCollection([], {model: ThangType})
+    @heroes.url = '/db/thang.type?view=heroes'
+    @heroes.setProjection ['original', 'name', 'heroClass']
+    @heroes.comparator = 'gems' # TODO: Random? Alphabetical? Something else?
+    @supermodel.loadCollection(@heroes, 'heroes')
```

Comment:
Doing this instead because otherwise any subsequent fetches (not that there are any, but for in general) wouldn't have the `project` otherwise:

```    @heroes = new ThangTypes()
    @heroes.setProjection ['original', 'name', 'heroClass']
    @heroes.comparator = 'gems' # TODO: Random? Alphabetical? Something else?
    @supermodel.trackRequest @heroes.fetchHeroes()```